File: Java/Csharp/AbstractFactory/AbstractFactory/ProductB1.cs
namespace AbstractFactory
{
    class ProductB1: IProductB
    {
        public string UsefulFunctionB()
        {
            return "The result of the product B1.";
        }

        public string AnotherUsefulFunctionB(IProductA collaborator)
        {
            var result = collaborator.UsefulFunctionA();
            return $"The result of the B1 collaborating with the ({result})";
        }
    }
}

File: PHP/Show/ProductB1.php
<?php
include_once('IProductB.php');

class ProductB1 implements IProductB
{

    public function usefulFunctionB(): string
    {
        return "The result of the product B1.";
    }

    public function anotherUsefulFunctionB(IProductA $collaborator): string
    {
        $result = $collaborator->usefulFunctionA();

        return "The result of the B1 collaborating with the ({$result})";
    }
}